cloud/gcp/modules/storage/main.tf

```diff
@@ -0,0 +1,75 @@
+resource "google_kms_key_ring" "keyring" {
+  name     = "civiform-keyring"
+  location = var.region
+}
+
+resource "google_kms_crypto_key" "storage_key" {
+  name            = "gcs-key"
+  key_ring        = google_kms_key_ring.keyring.id
+  rotation_period = "604800s" //7 days
+
+  lifecycle {
+    prevent_destroy = false
+  }
+}
+
+// Need the gcs service account name beforehand.
+data "google_storage_project_service_account" "gcs_account" {
+}
+
+// Allow gcs service account on key.
+resource "google_kms_crypto_key_iam_binding" "kms_binding" {
+  crypto_key_id = google_kms_crypto_key.storage_key.id
+  role          = "roles/cloudkms.cryptoKeyEncrypterDecrypter"
+  members = ["serviceAccount:${data.google_storage_project_service_account.gcs_account.email_address}"]
+}
+
+
+resource "google_storage_bucket" "file_storage" {
+  name          = "cloud-filestore-${var.application_name_postfix}"
+  location      = var.region
+  force_destroy = true
+
+  uniform_bucket_level_access = true
+
+  versioning {
+    enabled = true
+  }
+
+  encryption {
+    default_kms_key_name = google_kms_crypto_key.storage_key.id
+  }
```

Comment:
question (blocking): It looks like Cloud Storage offers some level of at-rest/server-side encryption by default. What's the benefit of using our own encryption key here?
https://cloud.google.com/storage/docs/encryption/default-keys

Reply:
@kunwarsingh1092 the GCP setup doesn't need to be an exact copy of the AWS environment. If using default encryption is also secure and also easier let's do that.